src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -326,6 +353,42 @@ private async Task ExecuteRequests(Hl7.Fhir.Model.Bundle responseBundle, Hl7.Fhi
             }
         }
 
+        private static void PopulateReferenceIdDictionary(IEnumerable<EntryComponent> bundleEntries, IDictionary<string, string> idDictionary)
+        {
+            foreach (EntryComponent entry in bundleEntries)
+            {
+                if (entry.Request.Method != HTTPVerb.POST)
+                {
```

Comment:
Is there a reason why we are handling this only for POST and not for PUT?  Does the reference resolution applies to only create scenarios and not for update?

Reply:
`PUT` requires there to be an explicit ID provided in the request. 

My assumption (could be incorrect) is that if you are referencing something that is the subject of a `PUT` you would have already updated the references to it to use that explicit ID provided in the request.